### ISSUE TITLE
[Windows test failure] rspec ./spec/bundler/dsl_spec.rb:210 # Bundler::Dsl#gemspec with a ruby platform keeps track of the ruby platforms in the dependency

### ISSUE THREAD

**Issue by janpio** [bundler/bundler#6887]
As described in https://github.com/bundler/bundler/issues/6868#issuecomment-451418573 currently 665 out of 2509 tests of bundler are failing on my Windows environment. I am working on fixing this.

---

Next one:

```
C:\Projects\bundler (windows -> origin)                                                                                                                             
λ ruby bin/rspec ./spec/bundler/dsl_spec.rb:210                                                                                                                     
Run options:                                                                                                                                                        
  include {:locations=>{"./spec/bundler/dsl_spec.rb"=>[210]}}                                                                                                       
  exclude {:ruby_repo=>false, :bundler=>"=< 2.0", :rubygems_master=>true, :git=>"=< 2.20.1", :rubygems=>"=< 2.6.14.1", :ruby=>"=< 2.4.4", :realworld=>true, :sudo=>t
                                                                                                                                                                    
Bundler::Dsl                                                                                                                                                        
  #gemspec                                                                                                                                                          
    with a ruby platform                                                                                                                                            
      keeps track of the ruby platforms in the dependency (FAILED - 1)                                                                                              
                                                                                                                                                                    
                                                                                                                                                                    
Retried examples: 0                                                                                                                                                 
                                                                                                                                                                    
Failures:                                                                                                                                                           
                                                                                                                                                                    
  1) Bundler::Dsl#gemspec with a ruby platform keeps track of the ruby platforms in the dependency                                                                  
     Failure/Error: expect(subject.dependencies.last.platforms).to eq(Bundler::Dependency::REVERSE_PLATFORM_MAP[Gem::Platform::RUBY])                               
                                                                                                                                                                    
       expected: [:ruby, :ruby_18, :ruby_19, :ruby_20, :ruby_21, :ruby_22, :ruby_23, :ruby_24, :ruby_25, :mri, :mri_18, :mri_19, :mri_20, :mri_21, :mri_22, :mri_23,
, :rbx, :truffleruby]                                                                                                                                               
            got: [:x64_mingw, :x64_mingw_20, :x64_mingw_21, :x64_mingw_22, :x64_mingw_23, :x64_mingw_24, :x64_mingw_25]                                             
                                                                                                                                                                    
       (compared using ==)                                                                                                                                          
                                                                                                                                                                    
       Diff:                                                                                                                                                        
       @@ -1,21 +1,8 @@                                                                                                                                             
       -[:ruby,                                                                                                                                                     
       - :ruby_18,                                                                                                                                                  
       - :ruby_19,                                                                                                                                                  
       - :ruby_20,                                                                                                                                                  
       - :ruby_21,                                                                                                                                                  
       - :ruby_22,                                                                                                                                                  
       - :ruby_23,                                                                                                                                                  
       - :ruby_24,                                                                                                                                                  
       - :ruby_25,                                                                                                                                                  
       - :mri,                                                                                                                                                      
       - :mri_18,                                                                                                                                                   
       - :mri_19,                                                                                                                                                   
       - :mri_20,                                                                                                                                                   
       - :mri_21,                                                                                                                                                   
       - :mri_22,                                                                                                                                                   
       - :mri_23,                                                                                                                                                   
       - :mri_24,                                                                                                                                                   
       - :mri_25,                                                                                                                                                   
       - :rbx,                                                                                                                                                      
       - :truffleruby]                                                                                                                                              
       +[:x64_mingw,                                                                                                                                                
       + :x64_mingw_20,                                                                                                                                             
       + :x64_mingw_21,                                                                                                                                             
       + :x64_mingw_22,                                                                                                                                             
       + :x64_mingw_23,                                                                                                                                             
       + :x64_mingw_24,                                                                                                                                             
       + :x64_mingw_25]                                                                                                                                             
                                                                                                                                                                    
     # ./spec/bundler/dsl_spec.rb:212:in `block (4 levels) in <top (required)>'                                                                                     
     # ./spec/spec_helper.rb:122:in `block (2 levels) in <top (required)>'                                                                                          
                                                                                                                                                                    
Finished in 5.31 seconds (files took 1.08 seconds to load)                                                                                                          
1 example, 1 failure                                                                                                                                                
                                                                                                                                                                    
Failed examples:                                                                                                                                                    
                                                                                                                                                                    
rspec ./spec/bundler/dsl_spec.rb:210 # Bundler::Dsl#gemspec with a ruby platform keeps track of the ruby platforms in the dependency                                
```

Looking at `Bundler::Dependency::REVERSE_PLATFORM_MAP` that is used to compare here, and the next test that mocks a response for jrubym it is quite obvious that here this test using the platform data to compare, which is different on Windows.

**Comment by janpio** [comment]
Fixed by adding simple `allow(Gem::Platform).to receive(:local).and_return(rb)` to the test. Will create PR.

**Comment by deivid-rodriguez** [comment]
Closed by #7542.
